src/java/org/jivesoftware/openfire/pubsub/PubSubEngine.java

```diff
@@ -543,7 +543,7 @@ private void subscribeNode(PubSubService service, IQ iq, Element childElement, E
             return;
         }
         // Check if the subscriber is an anonymous user
-        if (!UserManager.getInstance().isRegisteredUser(subscriberJID)) {
+        if (!UserManager.getInstance().isRegisteredUser(subscriberJID) && !isComponent(subscriberJID)) {
```

Comment:
Thanks; Java does to shortcuts with `&&` (not with `&`) so I've re-ordered the check